### PR TITLE
Make `Vector2i` values paired with `EDSCALE` be just `Vector2`

### PR DESCRIPTION
**Pull request by YeldhamDev**
If the editor's scale is set to a value lower than 1, `Vector2i` values of integer type paired with it will ignore it.

### CODE REVIEW

File: scene/gui/control.cpp [hunk 1]
```diff
@@ -1568,7 +1568,7 @@ void Control::set_custom_minimum_size(const Size2 &p_custom) {
 	if (p_custom == data.custom_minimum_size) {
 		return;
 	}
-	data.custom_minimum_size = p_custom;
+	data.custom_minimum_size = p_custom.round();
```
Comment (YuriSizov) [inline scene/gui/control.cpp:1571]: I still think that this method (and `set_fixed_icon_size` I guess) should take `Size2i` instead, then it's the responsibility of the caller to make sure the size provided is rounded the way they want it. `popup_centered` already takes `Size2i` for example.

**Review by YuriSizov — APPROVED** [review]
Aside from the note above, this should be good.

File: scene/gui/control.cpp (hunk 1 above)
Comment (YeldhamDev) [reply scene/gui/control.cpp:1571]: Sure, I just think it feels weird for just that `Control` function to accept `Size2i` while the others are still `Size2`.

Comment (YuriSizov) [reply scene/gui/control.cpp:1571]: Gotta start somewhere :) Also controls are pixel perfect, don't have any sub-pixels, right? So it makes sense here specifically.

Comment (YeldhamDev) [reply scene/gui/control.cpp]: Done.

**Comment by YuriSizov** [comment]
I guess tests don't like the integer sizes, not for the icons at least.

**Comment by YeldhamDev** [comment]
@YuriSizov Fixed. I forgot to change the getters.

**Comment by YeldhamDev** [comment]
@akien-mga I have no clue what file I need to make the changes.

File: modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs [hunk 2]
```diff
@@ -122,7 +122,7 @@ public override void _Ready()
         {
             base._Ready();
 
-            CustomMinimumSize = new Vector2(0, 228) * EditorScale;
+            CustomMinimumSize = new Vector2i(0, 228 * EditorScale);
```
Comment (akien-mga) [inline modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs:125]: Interesting GitHub feature for C# code. Why can't they do that for C++? :)
image

Anyway if C# wants to be super explicit I guess you might do:
```
new Vector2i(0, (int)(228.0 * EditorScale))
```

Or maybe @godotengine/dotnet have a better suggestion.

Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs]: About the GitHub feature, I'm pretty sure the dotnet action we use in CI is what reports those errors so it should be possible to do it for C++ if someone implemented it (see the `core.error` and the `core.warning` commands in https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions).

---

About the `CustomMinimumSize`, I don't really have a better suggestion other than maybe:

```csharp
new Vector2i(0, 228 * (int)EditorScale);
```

Comment (akien-mga) [reply modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs]: That one would be a problem as we want `228 * 1.5` to give `342`, not `228 * 1 = 228`.

**Review by akien-mga — APPROVED** [review]

File: modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs (hunk 2 above)
Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs]: You are right, I don't know where my head was. By the way you are missing the parenthesis around `int` in your latest changes.

File: modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs [hunk 3]
```diff
@@ -122,7 +122,7 @@ public override void _Ready()
         {
             base._Ready();
 
-            CustomMinimumSize = new Vector2(0, 228) * EditorScale;
+            CustomMinimumSize = new Vector2i(0, int(228 * EditorScale));
```
[the remark this answers is not in the document]
Comment (YeldhamDev) [reply modules/mono/editor/GodotTools/GodotTools/Build/MSBuildPanel.cs]: Done.

**Comment by YeldhamDev** [comment]
@akien-mga `--doctool` didn't update the types, so I had to do it manually.

**Comment by YuriSizov** [comment]
> @akien-mga `--doctool` didn't update the types, so I had to do it manually.

You probably need to change the property bindings from `Variant::VECTOR2` to `Variant::VECTOR2I`, but it's curious this doesn't fail the build.

**Comment by YeldhamDev** [comment]
@YuriSizov Fixed. Running `--doctool` after my manual fix also didn't change anything either, so something is definitely wrong there.

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/Control.xml [hunk 4]
```diff
@@ -972,7 +972,7 @@
 		<member name="clip_contents" type="bool" setter="set_clip_contents" getter="is_clipping_contents" default="false">
 			Enables whether rendering of [CanvasItem] based children should be clipped to this control's rectangle. If [code]true[/code], parts of a child which would be visibly outside of this control's rectangle will not be rendered and won't receive input.
 		</member>
-		<member name="custom_minimum_size" type="Vector2" setter="set_custom_minimum_size" getter="get_custom_minimum_size" default="Vector2(0, 0)">
+		<member name="custom_minimum_size" type="Vector2i" setter="set_custom_minimum_size" getter="get_custom_minimum_size" default="Vector2i(0, 0)">
```
Comment (aaronfranke) [inline doc/classes/Control.xml:975]: The title of your PR says you made Vector2i values be just Vector2, but here you are changing Vector2 to Vector2i. What was the reason for this change?

Comment (YeldhamDev) [reply doc/classes/Control.xml:975]: This was discussed with Yuri in the dev chat, but in a nutshell: It's good for those functions to always take those values as integers.

It changes nothing for the main objective of this PR.

**Pull request merged**